ethcore/wasm/src/lib.rs

```diff
@@ -103,15 +103,25 @@ impl vm::Vm for WasmInterpreter {
 	fn exec(&mut self, params: ActionParams, ext: &mut dyn vm::Ext) -> vm::Result<GasLeft> {
 		let is_create = ext.is_create();
 
-		let parser::ParsedModule { module, code, data } = parser::payload(
-			&params,
-			ext.schedule().wasm(),
-			if is_create {
-				Some(subst_main_call)
-			} else {
-				None
-			},
-		)?;
```

Comment:
this bit was refactored into two steps because injecting a mut closure capturing mut locals (i.e. `did_subst_main`) is way more mut that I'm comfortable with.